src/Parser/DeathKnight/Blood/Modules/Features/BoneShield.js

```diff
@@ -0,0 +1,135 @@
+import React from 'react';
+import Analyzer from 'Parser/Core/Analyzer';
+import Combatants from 'Parser/Core/Modules/Combatants';
+import SPELLS from 'common/SPELLS';
+import SpellIcon from 'common/SpellIcon';
+import { formatNumber, formatPercentage } from 'common/format';
+import StatisticBox, { STATISTIC_ORDER } from 'Main/StatisticBox';
+import StatTracker from 'Parser/Core/Modules/StatTracker';
+
+class BoneShield extends Analyzer {
+
+  static dependencies = {
+    combatants: Combatants,
+    statTracker: StatTracker,
+  };
+
+  SS_DR = 0.08;
+  BONE_SHIELD_DR = 0.16;
+
+  totalDamageTaken = 0;
+
+  hasSS = false;
+  hasSD = false;
+
+  totalCrit = 0;
+  totalCritChecks = 0;
+
+  boneShieldDR = 0.16;
+  ssDR = 0;
+
+
+  on_initialized() {
+    this.hasSS = this.combatants.selected.traitsBySpellId[SPELLS.SKELETAL_SHATTERING_TRAIT.id];
+    this.hasSD = this.combatants.selected.hasTalent(SPELLS.SPECTRAL_DEFLECTION_TALENT.id);
+  }
+
+  get boneShieldAbsorbTooltip() {
+    if (this.hasSS) {
+      const avgCrit = this.totalCrit / this.totalCritChecks;
+      this.ssDR = avgCrit * this.SS_DR;
+      this.boneShieldDR = this.BONE_SHIELD_DR + this.ssDR;
+    }
+
+    return formatNumber(this.totalDamageTaken * this.boneShieldDR) + " Bone Shield Absorb<br>";
+  }
+
+  get skeletalShatteringTooltip() {
+    if (this.hasSS) {
+      return "On average, Skeletal Shattering would have contributed " + 
+        formatPercentage((this.totalDamageTaken * this.ssDR) / (this.totalDamageTaken * this.boneShieldDR)) +
+        "% (" + formatNumber(this.totalDamageTaken * this.ssDR) + ") to this.<br>";
+    } else {
+      return "";
+    }
+  }
+
+
+
+  on_toPlayer_damage(event) {
+
+    if (!this.combatants.selected.hasBuff(SPELLS.BONE_SHIELD.id)) {
+      return;
+    }
+
+    this.totalDamageTaken += event.amount + event.absorbed;
```

Comment:
You're right about the crit calculation, its more robust against spike-damage while a crit-buff is up.
You're also right about having to use premitigated damage instead of actual damage taken.
Updated the calculations.